### PR TITLE
Add expression segment for subclass.

### LINKED ISSUE #11127: Some subclass of `ExpressionSegment` have not been parsed
The expression of `ExpressionOrderByItemSegment`  and `ExpressionProjectionSegment` contains only String type field, which cannot be converted to calcite SqlNode.

Then expression field should be ast object.

### PR DESCRIPTION
**Pull request by tuichenchuxin**
Add expression segment for subclass.

Fixes #11127 

Changes proposed in this pull request:
- Add ExpressionSegment filed for ExpressionProjectionSegment.
- Add ExpressionSegment filed for ExpressionOrderByItemSegment.

### CODE REVIEW

**Comment by strongduanmu** [comment]
@tuichenchuxin Please solve the code conflict first.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLStatementSQLVisitor.java [hunk 1]
```diff
@@ -886,6 +886,8 @@ public final ASTNode visitOrderByItem(final OrderByItemContext ctx) {
             ASTNode expr = visitExpr(ctx.expr());
             if (expr instanceof ColumnSegment) {
                 return new ColumnOrderByItemSegment((ColumnSegment) expr, orderDirection);
+            } else if (expr instanceof ExpressionSegment) {
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLStatementSQLVisitor.java:889]: @tuichenchuxin Can we merge this logic with the following logic?

**Review by strongduanmu — COMMENTED** [review]
@tuichenchuxin Please add test cases related to parse test.

**Comment by tuichenchuxin** [comment]
This pull request is not reasonable, i'll close it.

**Pull request closed without merging**